### ISSUE TITLE
Pin most recent @steemitblog post to top of newsfeed

### ISSUE THREAD

**Issue by relativityboy** [steemit/condenser#3081]
Steem has undergone many changes of late, and as a user I want to know what's happening. Pin the most recent `@steemitblog` post to the top of `/new` or `/trending` for a period of time _so I can be informed_.

**AC**
- [ ] determine which feed to pin the most recent post to.  Add to `# Decisions` comment.
- [ ] determine how long the post should be pinned. Add to `# Decisions` comment.
- [ ] pin the post in the place and time determined by the previous two AC

**Comment by jredbeard** [comment]
# Decisions
* It won't be up to condenser to decide which posts to pin, nor how long the post should be pinned. It's preferred to handle this part manually.
* Condenser should read in a json file at startup from a URL that determines the pinned posts. If the file is not available or the json contains an empty set, 'pinned posts' will not be displayed.
* The workflow for updating the list of pinned posts will be to simply update the json file, then redeploy condenser so that it pulls in the latest file.
* Pinned posts should show up on `/new`, `/trending`, and `/@user/feed`

An example of what the json file could look like:

```
{
  "pinned": [
    "/tag/username/perm-link-here",
    "/tag/username/another-perm-link-here"
  ]
}
```
Or for a no pinned posts:
```
{
  "pinned": [
  ]
}
```

**Comment by relativityboy** [comment]
The above is an excellent starting point. 

* We're going  to add an expiration date/time to each post's json... 
* Reading the json file in from a url at startup sounds like a performance killer on the client-side that'll result in the 'jumping ui' experience. We want the json data to be built right into condenser.

**Comment by bnchdrff** [comment]
i'm pretty sure justin means that the condenser server-side process will read the json file at startup

**Comment by jredbeard** [comment]
Yes @bnchdrff you are correct - the intention is to read the file in once server-side at startup. Clients would not be pulling in data from the json file directly. The information would then live in memory until condenser is restarted and the file is pulled in again.

Make sure @pkattera wants an expiration - I think that the intention is to have this be 100% manual. If we want to remove the pinned items, we edit the json file to remove them.

**Comment by pkattera** [comment]
I like the idea of combining an automatic expiration (say 48 hours) which can be overridden manually by marketing. However, I think it adds complexity. For example, if marketing wanted to keep a post pinned for another day, how would they do so within the json file? We'd probably need another parameter to say 'no-expiry'. 

Anyway, I think the simplest solution for now is to just manually adjust via the json file.

**Comment by jredbeard** [comment]
Closed by #3114  - thanks @rjungemann
